### PR TITLE
add condition to canStop for runStrategy

### PR DESCRIPTION
**Pull request by tjjh89017**
if there is no running attr, we should consider runStrategy

Signed-off-by: Date Huang <date.huang@suse.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

VM with RerunOnFailure in off state should not have stop action.

**Solution:**

runStrategy RerunOnFailure and Manula should consider VMI status also.
if VMI isn't existing or VMI is not running, don't allow stop action.

**Related Issue:** https://github.com/harvester/harvester/issues/1882

**Test plan:**

1. Create a VM with `runStrategy: RerunOnFailure`
2. Open VM console and poweroff VM via console command (`sudo poweroff`)
3. UI will show Off state (if UI is fixed already)
4. UI will not show Stop action in action menu.

### CODE REVIEW

**Review by weihanglo — APPROVED** [review]
LGTM for the backend part. I guess we can merge this first when getting one more approval.

**Review by bk201 — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**
